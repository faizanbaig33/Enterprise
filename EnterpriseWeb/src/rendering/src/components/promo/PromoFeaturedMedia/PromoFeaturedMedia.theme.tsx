import { ThemeFile } from 'lib/context/ThemeContext';
import classNames from 'classnames';

export const PromoFeaturedMediaTheme: ThemeFile = {
  aw: {
    classes: {
      imageDivClass: 'order-first col-span-12',
      captionClass: 'hidden',
      leftContainerClass: 'col-span-9 my-auto mt-0 flex flex-col self-center px-xl py-s pr-xxl',
      subheadlineClass: 'text-theme-text text-small uppercase font-bold mb-2',
      headlineClass: 'text-theme-text text-xs font-heavy mb-2',
      bodyClass: 'text-theme-body mb-4',
      buttonGroupClass: {
        wrapper: classNames(
          'col-span-12',
          'flex',
          'flex-col',
          'lg:flex-row',
          'items-start',
          'text-xs',
          'font-heavy'
        ),
        cta1Classes: classNames('lg:mr-3 lg:first:ml-0 lg:last:mr-0', 'bg-theme-bg'),
        cta2Classes: '',
      },
      rightContainerClass: 'col-span-3 my-auto mt-0 flex flex-col self-center px-l py-s pr-xl',
      highlightContainerClass: '',
      highlightTitleClass:
        'text-theme-text text-small uppercase font-bold py-2 border-gray border-t',
      highlightDescriptionClass: 'text-theme-body text-small mb-m',
    },
  },
  rba: {
    classes: {
      imageDivClass: 'order-first col-span-12',
      captionClass: 'hidden',
      leftContainerClass:
        'col-span-6 my-auto mt-0 flex flex-col self-center px-xl py-s pr-xxl pt-xxs',
      subheadlineClass: 'text-theme-text text-small uppercase font-bold mb-2',
      headlineClass: 'text-theme-text text-xs font-bold mb-2',
      bodyClass: 'text-theme-body mb-4',
      buttonGroupClass: {
        wrapper: classNames(
          'col-span-12',
          'flex',
          'flex-col',
          'lg:flex-row',
          'items-start',
          'text-xs',
          'font-heavy'
        ),
        cta1Classes: classNames('lg:mr-3 lg:first:ml-0 lg:last:mr-0', 'bg-theme-bg'),
        cta2Classes: '',
      },
      rightContainerClass:
        'col-span-6 my-auto mx-auto flex flex-row self-center px-l py-s pr-xl pt-xxs justify-between',
      highlightContainerClass: 'mr-xl',
      highlightTitleClass:
        'text-theme-body text-small font-heavy uppercase border-theme-btn-bg-hover border-t-4 border-top',
      highlightDescriptionClass: 'text-theme-text font-bold',
    },
  },
};
