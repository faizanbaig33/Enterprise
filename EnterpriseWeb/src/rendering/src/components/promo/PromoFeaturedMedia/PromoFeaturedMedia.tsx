import { Component } from 'src/helpers/Component';
import { getEnum } from 'lib/utils';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { MediaPrimary } from 'src/helpers/Media';
import { Headline } from 'src/helpers/Headline';
import { withDatasourceCheck, Text } from '@sitecore-jss/sitecore-jss-nextjs';
import { useTheme } from 'lib/context/ThemeContext';
import { PromoFeaturedMediaTheme } from './PromoFeaturedMedia.theme';
import { Subheadline } from 'src/helpers/Subheadline';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { RichTextWrapper } from 'src/helpers/RichTextWrapper';
import { ButtonGroup } from 'src/helpers/ButtonGroup';
import { ComponentBackgroundVariants } from 'src/helpers/Component/Component';

export type PromoFeaturedMediaProps =
  Feature.EnterpriseWeb.Components.Promo.PromoFeaturedMedia.PromoFeaturedMedia;

const PromoFeaturedMedia = (props: PromoFeaturedMediaProps): JSX.Element => {
  const { themeData } = useTheme(PromoFeaturedMediaTheme);
  const style = getEnum<ComponentBackgroundVariants>(props.fields?.backgroundColor) ?? 'white';

  return (
    <Component
      variant="lg"
      backgroundVariant={style}
      dataComponent="promo/promofeaturedmedia"
      {...props}
    >
      <div className={themeData.classes.imageDivClass}>
        <MediaPrimary {...props} />
      </div>
      <div className={themeData.classes.leftContainerClass}>
        <Subheadline classes={themeData.classes.subheadlineClass} {...props} />
        <div className={themeData.classes.bodyClass}>
          <RichTextWrapper field={props.fields?.primaryImageCaption} />
        </div>
        {props.fields?.headlineText?.value &&
          props.fields?.body?.value &&
          props.fields?.cta1Link?.value?.text && (
            <div>
              <Headline classes={themeData.classes.headlineClass} {...props} />
              <BodyCopy classes={themeData.classes.bodyClass} {...props} />
              <ButtonGroup classes={themeData.classes.buttonGroupClass} {...props} />
            </div>
          )}
      </div>
      <div className={themeData.classes.rightContainerClass}>
        {props.fields?.highlightTitle1?.value && props.fields?.highlightDescription1?.value && (
          <div className={themeData.classes.highlightContainerClass}>
            <div className={themeData.classes.highlightTitleClass}>
              <Text tag={'h3'} field={props.fields?.highlightTitle1} />
            </div>
            <RichTextWrapper
              field={props.fields?.highlightDescription1}
              className={themeData.classes.highlightDescriptionClass}
            />
          </div>
        )}
        {props.fields?.highlightTitle2?.value && props.fields?.highlightDescription2?.value && (
          <div className={themeData.classes.highlightContainerClass}>
            <div className={themeData.classes.highlightTitleClass}>
              <Text tag={'h3'} field={props.fields?.highlightTitle2} />
            </div>
            <RichTextWrapper
              field={props.fields?.highlightDescription2}
              className={themeData.classes.highlightDescriptionClass}
            />
          </div>
        )}
        {props.fields?.highlightTitle3?.value && props.fields?.highlightDescription3?.value && (
          <div className={themeData.classes.highlightContainerClass}>
            <div className={themeData.classes.highlightTitleClass}>
              <Text tag={'h3'} field={props.fields?.highlightTitle3} />
            </div>
            <RichTextWrapper
              field={props.fields?.highlightDescription3}
              className={themeData.classes.highlightDescriptionClass}
            />
          </div>
        )}
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<PromoFeaturedMediaProps>(PromoFeaturedMedia);
