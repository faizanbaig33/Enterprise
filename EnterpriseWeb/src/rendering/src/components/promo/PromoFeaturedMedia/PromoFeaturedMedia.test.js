// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import PromoFeaturedMedia from './PromoFeaturedMedia';
import defaultData from './PromoFeaturedMedia.mock-data';
import blackData from './PromoFeaturedMedia.mock-data-black';
import grayData from './PromoFeaturedMedia.mock-data-gray';
import videoData from './PromoFeaturedMedia.mock-data-video';

it('renders correctly', () => {
  const component = snapshot(PromoFeaturedMedia, { componentProps: defaultData });
  hasDataComponent(component, 'promo/promofeaturedmedia');
});

it('black renders correctly', () => {
  const component = snapshot(PromoFeaturedMedia, { componentProps: blackData });
  hasDataComponent(component, 'promo/promofeaturedmedia');
});

it('gray renders correctly', () => {
  const component = snapshot(PromoFeaturedMedia, { componentProps: grayData });
  hasDataComponent(component, 'promo/promofeaturedmedia');
});

it('video renders correctly', () => {
  const component = snapshot(PromoFeaturedMedia, { componentProps: videoData });
  hasDataComponent(component, 'promo/promofeaturedmedia');
});
