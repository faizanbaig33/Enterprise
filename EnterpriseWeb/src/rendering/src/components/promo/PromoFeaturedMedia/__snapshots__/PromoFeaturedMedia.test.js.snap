// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`black renders correctly 1`] = `
<section
  data-component="promo/promofeaturedmedia"
  id="Promo Featured Media"
>
  <div
    class="lg:mx-auto md:max-w-screen-lg theme-black bg-theme-bg"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s"
    >
      <div
        class="order-first col-span-12"
      >
        <div
          class="relative hidden md:block aspect-video   "
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 42.857142857142854% 0px 0px 0px;"
            />
            <img
              alt="Unsplash Source"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="relative w-full md:hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 42.857142857142854% 0px 0px 0px;"
            />
            <img
              alt="Unsplash Source"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="caption"
        >
          <div
            class="text-theme-text ml:text-secondary mt-xxs italic text-left mb-s font-sans text-sm-xxs md:text-caption"
          >
            Description of product with 
            <a
              href="google.com"
            >
              link
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-span-9 my-auto mt-0 flex flex-col self-center px-xl py-s pr-xxl"
      >
        <div
          class="font-sans text-theme-text text-small uppercase font-bold mb-2"
        >
          <h3>
            Featured Media Sub Heading Text
          </h3>
        </div>
        <div
          class="text-theme-body mb-4"
        >
          <div
            class="body-copy"
          >
            <div>
              Description of product with 
              <a
                href="google.com"
              >
                link
              </a>
            </div>
          </div>
        </div>
        <div>
          <div
            class="items-top font-sans  text-theme-text text-xs font-heavy mb-2"
          >
            <h3>
              Featured Media Headline Text
            </h3>
          </div>
          <div
            class="text-theme-body mb-4"
          >
            <div
              class="body-copy"
            >
              <div>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit lectus, et sagittis nibh risus vel metus. Quisque eu ornare ante, et gravida mauris.
              </div>
            </div>
          </div>
          <div
            class="col-span-12 flex flex-col lg:flex-row items-start text-xs font-heavy mb-s flex items-start md:flex-row md:items-center md:space-x-4"
          >
            <a
              class="flex w-fit items-center whitespace-nowrap font-sans text-text-link font-heavy text-theme-text hover:underline hover:decoration-secondary hover:underline-offset-8 disabled:border-gray disabled:text-gray lg:mr-3 lg:first:ml-0 lg:last:mr-0 bg-theme-bg mr-m"
              href="/"
              rel="noopener noreferrer"
              role="link"
              tabindex="0"
              target="_blank"
              title=""
            >
              Learn More
              <span
                class="ml-xxs"
              />
              <span
                class="sr-only"
              >
                 (Opens in a new tab)
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-span-3 my-auto mt-0 flex flex-col self-center px-l py-s pr-xl"
      >
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight One
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight Two
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight Three
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`gray renders correctly 1`] = `
<section
  data-component="promo/promofeaturedmedia"
  id="Promo Featured Media"
>
  <div
    class="lg:mx-auto md:max-w-screen-lg theme-gray bg-light-gray"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s"
    >
      <div
        class="order-first col-span-12"
      >
        <div
          class="relative hidden md:block aspect-video   "
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 42.857142857142854% 0px 0px 0px;"
            />
            <img
              alt="Unsplash Source"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="relative w-full md:hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 42.857142857142854% 0px 0px 0px;"
            />
            <img
              alt="Unsplash Source"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="caption"
        >
          <div
            class="text-theme-text ml:text-secondary mt-xxs italic text-left mb-s font-sans text-sm-xxs md:text-caption"
          >
            Description of product with 
            <a
              href="google.com"
            >
              link
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-span-9 my-auto mt-0 flex flex-col self-center px-xl py-s pr-xxl"
      >
        <div
          class="font-sans text-theme-text text-small uppercase font-bold mb-2"
        >
          <h3>
            Featured Media Sub Heading Text
          </h3>
        </div>
        <div
          class="text-theme-body mb-4"
        >
          <div
            class="body-copy"
          >
            <div>
              Description of product with 
              <a
                href="google.com"
              >
                link
              </a>
            </div>
          </div>
        </div>
        <div>
          <div
            class="items-top font-sans  text-theme-text text-xs font-heavy mb-2"
          >
            <h3>
              Featured Media Headline Text
            </h3>
          </div>
          <div
            class="text-theme-body mb-4"
          >
            <div
              class="body-copy"
            >
              <div>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit lectus, et sagittis nibh risus vel metus. Quisque eu ornare ante, et gravida mauris.
              </div>
            </div>
          </div>
          <div
            class="col-span-12 flex flex-col lg:flex-row items-start text-xs font-heavy mb-s flex items-start md:flex-row md:items-center md:space-x-4"
          >
            <a
              class="flex w-fit items-center whitespace-nowrap font-sans text-text-link font-heavy text-theme-text hover:underline hover:decoration-secondary hover:underline-offset-8 disabled:border-gray disabled:text-gray lg:mr-3 lg:first:ml-0 lg:last:mr-0 bg-theme-bg mr-m"
              href="/"
              role="link"
              tabindex="0"
              target="_blank"
              title=""
            >
              Learn More
              <span
                class="ml-xxs"
              />
              <span
                class="sr-only"
              >
                 (Opens in a new tab)
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-span-3 my-auto mt-0 flex flex-col self-center px-l py-s pr-xl"
      >
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight One
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight Two
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight Three
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly 1`] = `
<section
  data-component="promo/promofeaturedmedia"
  id="Promo Featured Media"
>
  <div
    class="lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s"
    >
      <div
        class="order-first col-span-12"
      >
        <div
          class="relative hidden md:block aspect-video   "
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 42.857142857142854% 0px 0px 0px;"
            />
            <img
              alt="Unsplash Source"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="relative w-full md:hidden"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 42.857142857142854% 0px 0px 0px;"
            />
            <img
              alt="Unsplash Source"
              data-nimg="responsive"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="caption"
        >
          <div
            class="text-theme-text ml:text-secondary mt-xxs italic text-left mb-s font-sans text-sm-xxs md:text-caption"
          >
            Description of product with 
            <a
              href="google.com"
            >
              link
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-span-9 my-auto mt-0 flex flex-col self-center px-xl py-s pr-xxl"
      >
        <div
          class="font-sans text-theme-text text-small uppercase font-bold mb-2"
        >
          <h3>
            Featured Media Sub Heading Text
          </h3>
        </div>
        <div
          class="text-theme-body mb-4"
        >
          <div
            class="body-copy"
          >
            <div>
              Description of product with 
              <a
                href="google.com"
              >
                link
              </a>
            </div>
          </div>
        </div>
        <div>
          <div
            class="items-top font-sans  text-theme-text text-xs font-heavy mb-2"
          >
            <h3>
              Featured Media Headline Text
            </h3>
          </div>
          <div
            class="text-theme-body mb-4"
          >
            <div
              class="body-copy"
            >
              <div>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit lectus, et sagittis nibh risus vel metus. Quisque eu ornare ante, et gravida mauris.
              </div>
            </div>
          </div>
          <div
            class="col-span-12 flex flex-col lg:flex-row items-start text-xs font-heavy mb-s flex items-start md:flex-row md:items-center md:space-x-4"
          >
            <a
              class="flex w-fit items-center whitespace-nowrap font-sans text-text-link font-heavy text-theme-text hover:underline hover:decoration-secondary hover:underline-offset-8 disabled:border-gray disabled:text-gray lg:mr-3 lg:first:ml-0 lg:last:mr-0 bg-theme-bg mr-m"
              href="/"
              role="link"
              tabindex="0"
              target="_blank"
              title=""
            >
              Learn More
              <span
                class="ml-xxs"
              />
              <span
                class="sr-only"
              >
                 (Opens in a new tab)
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-span-3 my-auto mt-0 flex flex-col self-center px-l py-s pr-xl"
      >
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight One
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight Two
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight Three
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`video renders correctly 1`] = `
<section
  data-component="promo/promofeaturedmedia"
  id="Promo Featured Media"
>
  <div
    class="lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s"
    >
      <div
        class="order-first col-span-12"
      >
        <div
          style="position: relative; overflow: hidden; width: 100%; padding-top: 56.25%;"
        >
          <iframe
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowfullscreen=""
            id="YWkfIwORD-M"
            loading="lazy"
            name="ytwrapper"
            src="https://www.youtube.com/embed/YWkfIwORD-M?enablejsapi=1&loop=1&controls=1&autoplay=0&disablekb=0&modestbranding=0&mute=true&playlist=YWkfIwORD-M"
            style="position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; width: 100%; height: 100%;"
            title="Youtube Video"
          />
        </div>
      </div>
      <div
        class="col-span-9 my-auto mt-0 flex flex-col self-center px-xl py-s pr-xxl"
      >
        <div
          class="font-sans text-theme-text text-small uppercase font-bold mb-2"
        >
          <h3>
            Featured Media Sub Heading Text
          </h3>
        </div>
        <div
          class="text-theme-body mb-4"
        >
          <div
            class="body-copy"
          >
            <div>
              Description of product with 
              <a
                href="google.com"
              >
                link
              </a>
            </div>
          </div>
        </div>
        <div>
          <div
            class="items-top font-sans  text-theme-text text-xs font-heavy mb-2"
          >
            <h3>
              Featured Media Headline Text
            </h3>
          </div>
          <div
            class="text-theme-body mb-4"
          >
            <div
              class="body-copy"
            >
              <div>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit lectus, et sagittis nibh risus vel metus. Quisque eu ornare ante, et gravida mauris.
              </div>
            </div>
          </div>
          <div
            class="col-span-12 flex flex-col lg:flex-row items-start text-xs font-heavy mb-s flex items-start md:flex-row md:items-center md:space-x-4"
          >
            <a
              class="flex w-fit items-center whitespace-nowrap font-sans text-text-link font-heavy text-theme-text hover:underline hover:decoration-secondary hover:underline-offset-8 disabled:border-gray disabled:text-gray lg:mr-3 lg:first:ml-0 lg:last:mr-0 bg-theme-bg mr-m"
              href="/"
              rel="noopener noreferrer"
              role="link"
              tabindex="0"
              target="_blank"
              title=""
            >
              Learn More
              <span
                class="ml-xxs"
              />
              <span
                class="sr-only"
              >
                 (Opens in a new tab)
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-span-3 my-auto mt-0 flex flex-col self-center px-l py-s pr-xl"
      >
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight One
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight Two
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="text-theme-text text-small uppercase font-bold py-2 border-gray border-t"
          >
            <h3>
              Highlight Three
            </h3>
          </div>
          <div
            class="body-copy"
          >
            <div
              class="text-theme-body text-small mb-m"
            >
              Lorem ipsum
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
